// @crowdin/crowdin-api-client/out/styleGuides/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StyleGuides = void 0;
const core_1 = require("../core");
class StyleGuides extends core_1.CrowdinApi {
    /**
     * @param options optional parameters for the request
     * @see https://developer.crowdin.com/api/v2/#operation/api.style-guides.getMany
     */
    listStyleGuides(options) {
        let url = `${this.url}/style-guides`;
        url = this.addQueryParam(url, 'orderBy', options === null || options === void 0 ? void 0 : options.orderBy);
        url = this.addQueryParam(url, 'userId', options === null || options === void 0 ? void 0 : options.userId);
        return this.getList(url, options === null || options === void 0 ? void 0 : options.limit, options === null || options === void 0 ? void 0 : options.offset);
    }
    /**
     * @param request request body
     * @see https://developer.crowdin.com/api/v2/#operation/api.style-guides.post
     */
    createStyleGuide(request) {
        const url = `${this.url}/style-guides`;
        return this.post(url, request, this.defaultConfig());
    }
    /**
     * @param styleGuideId style guide identifier
     * @see https://developer.crowdin.com/api/v2/#operation/api.style-guides.get
     */
    getStyleGuide(styleGuideId) {
        const url = `${this.url}/style-guides/${styleGuideId}`;
        return this.get(url, this.defaultConfig());
    }
    /**
     * @param styleGuideId style guide identifier
     * @see https://developer.crowdin.com/api/v2/#operation/api.style-guides.delete
     */
    deleteStyleGuide(styleGuideId) {
        const url = `${this.url}/style-guides/${styleGuideId}`;
        return this.delete(url, this.defaultConfig());
    }
    /**
     * @param styleGuideId style guide identifier
     * @param request request body
     * @see https://developer.crowdin.com/api/v2/#operation/api.style-guides.patch
     */
    editStyleGuide(styleGuideId, request) {
        const url = `${this.url}/style-guides/${styleGuideId}`;
        return this.patch(url, request, this.defaultConfig());
    }
}
exports.StyleGuides = StyleGuides;
